app/controllers/api/messaging.php

```
@@ -1744,21 +1914,212 @@
+    });
+
+App::patch('/v1/messaging/messages/sms/:messageId')
+    ->desc('Update an sms.')
+    ->groups(['api', 'messaging'])
+    ->label('audits.event', 'messages.update')
+    ->label('audits.resource', 'messages/{response.$id}')
+    ->label('scope', 'messages.write')
+    ->label('sdk.auth', [APP_AUTH_TYPE_ADMIN, APP_AUTH_TYPE_KEY])
+    ->label('sdk.namespace', 'messaging')
+    ->label('sdk.method', 'updateSMS')
+    ->label('sdk.description', '/docs/references/messaging/update-email.md')
+    ->label('sdk.response.code', Response::STATUS_CODE_OK)
+    ->label('sdk.response.type', Response::CONTENT_TYPE_JSON)
+    ->label('sdk.response.model', Response::MODEL_MESSAGE)
+    ->param('messageId', '', new UID(), 'Message ID.')
+    ->param('to', [], new ArrayList(new Text(Database::LENGTH_KEY)), 'List of Topic IDs or List of User IDs or List of Target IDs.', true)
+    ->param('description', '', new Text(256), 'Description for Message.', true)
+    ->param('content', '', new Text(64230), 'Email Content.', true)
+    ->param('status', '', new WhiteList(['draft', 'processing']), 'Message Status. Value must be either draft or processing.', true)
+    ->param('deliveryTime', null, new DatetimeValidator(requireDateInFuture: true), 'Delivery time for message in ISO 8601 format. DateTime value must be in future.', true)
+    ->inject('dbForProject')
+    ->inject('project')
+    ->inject('messaging')
+    ->inject('response')
+    ->action(function (string $messageId, array $to, string $description, string $content, string $status, ?string $deliveryTime, Database $dbForProject, Document $project, Messaging $messaging, Response $response) {
+        $message = $dbForProject->getDocument('messages', $messageId);
+
+        if ($message->isEmpty()) {
+            throw new Exception(Exception::MESSAGE_NOT_FOUND);
+        }
+
+        if ($message->getAttribute('status') === 'sent') {
+            throw new Exception(Exception::MESSAGE_ALREADY_SENT);
+        }
+
```

Let's also throw an exception if it's after the `deliveryTime` to avoid triggering duplicates in the time between triggering delivery and the message actually being delivered

---

done this